batch/src/main/scala/uk/gov/ons/addressindex/models/EsDocument.scala

```diff
@@ -28,8 +28,12 @@ abstract class EsDocument {
     val premsAndThoroughfare = (premises, thoroughfares) match {
       case (sub_build :: build :: Some(number) :: Nil, Some(thorough_first) :: thorough_rest) =>
         sub_build :: build :: poBox :: Some(number + " " + thorough_first) :: thorough_rest
+      case (sub_build :: build :: Some(number) :: Nil, None :: thorough_rest) =>
+        sub_build :: build :: poBox :: Some(number + " " + thorough_rest.headOption.getOrElse(Some("")).getOrElse("")) :: Nil
       case (sub_build :: Some(build@startsWithNumber()) :: None :: Nil, Some(thorough_first) :: thorough_rest) =>
         sub_build :: poBox :: Some(build + " " + thorough_first) :: thorough_rest
+      case (sub_build :: Some(build@startsWithNumber()) :: None :: Nil, None :: thorough_rest) =>
+        sub_build :: poBox :: Some(build + " " + thorough_rest.headOption.getOrElse(Some("")).getOrElse("")) :: Nil
```

Comment:
Will check the conditions Thomas pointed out.
Also I just spotted some cases where the NAG address has the unwanted comma instead of the PAF (example 20 Addington Court, Horseguards Exeter) so I'll investigate that too and do a new push.

Reply:
The NAG code appears to be correct but the test address has fields wrongly populated, either due to a data glitch on AddressBase or a different problem with the Spark job. Will create a new ticket to investigate this one.
The empty list changes suggested didn't give the correct results.
So will merge as is.